src/main/java/org/variantsync/diffdetective/analysis/PatchAnalysis.java

```diff
@@ -0,0 +1,74 @@
+package org.variantsync.diffdetective.analysis;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.tinylog.Logger;
+import org.variantsync.diffdetective.editclass.EditClass;
+import org.variantsync.diffdetective.editclass.proposed.ProposedEditClasses;
+import org.variantsync.diffdetective.metadata.EditClassCount;
+import org.variantsync.diffdetective.util.CSV;
+import org.variantsync.diffdetective.util.FileUtils;
+import org.variantsync.diffdetective.util.IO;
+
+public class PatchAnalysis implements Analysis.Hooks {
```

Comment:
I see why you chose this name but it is misleading. This analyses counts only edit class occurences so name it `EditClassOccurenceAnalysis`?

Reply:
This would probably also mean to rename the `PatchStatistics` to something like `PatchEditClassRecord`?

Reply:
Sounds reasonable.

Because of that new name, I just spotted duplicate code: `EditClassValidation.analyzeDiffTree` and `PatchAnalysis.analyzeDiffTree` do the same thing except `PatchAnalysis.analyzeDiffTree` also stored the edit class occurrences in patch granularity. Should we do something abut this or is this acceptable?

Reply:
As the name `EditClassCounts` was overloaded (in `metadata` and `analysis`) I merged `EditClassCounts` into `PatchStatistics` and further simplified `EditClassOccurenceAnalysis` by inlining the result. It now also uses `AnalysisStrategy` for the export.

Reply:
I don't remember all the details here. I wouldn't be surprised If introduced quite some duplicate code here. Refactoring is fine and appreciated as long as it indeed is a refactoring.